parts/k8s/cloud-init/masternodecustomdata.yml

```diff
@@ -395,11 +395,17 @@ MASTER_CONTAINER_ADDONS_PLACEHOLDER
     # Redirect ILB (4443) traffic to port 443 (ELB) in the prerouting chain
     iptables -t nat -A PREROUTING -p tcp --dport 4443 -j REDIRECT --to-port 443
 {{end}}
-
     sed -i "s|<img>|{{WrapAsParameter "kubernetesAddonManagerSpec"}}|g" /etc/kubernetes/manifests/kube-addon-manager.yaml
+{{if IsKubernetesVersionGe "1.17.0-alpha.1"}}
+    sed -i "s|<img>|{{WrapAsParameter "kubeAPIServerSpec"}}|g" /etc/kubernetes/manifests/kube-apiserver.yaml
+    sed -i "s|<img>|{{WrapAsParameter "kubeControllerManagerSpec"}}|g" /etc/kubernetes/manifests/kube-controller-manager.yaml
+    sed -i "s|<img>|{{WrapAsParameter "kubeSchedulerSpec"}}|g" /etc/kubernetes/manifests/kube-scheduler.yaml
+    sed -i "s|\"/hyperkube\", ||g" /etc/kubernetes/manifests/kube-*.yaml
```

Comment:
can we just use 1.17-specific yamls with no hyperkube reference so we can omit this sed replacement?

Reply:
Sure, that's another approach, although we don't follow the version -> subdirectory convention in `parts/k8s/manifests` currently, just in addons. 

I was thinking of this `sed` as a temporary hack because the end goal is to replace hyperkube usage in all supported versions, at which point we can just fix the command in the manifests and drop the `sed`.

Reply:
Yeah, all good with iteratation.

Reply:
I went with 1.17-specific yamls as you suggested, because I'm not sure we can make a wholesale change off of hyperkube. We expose `customHyperkubeImage` in KubernetesConfig, so there's a backward-compatibility issue there.